### PR TITLE
Add Simple Message data & msg structures for Motoman 'Single IO' extensions

### PR DESCRIPTION
**Pull request by gavanderhoorn**
As per subject.

C++ classes for the (Motoman) `READ_SINGLE_IO`, `READ_SINGLE_IO_REPLY`, `WRITE_SINGLE_IO` and `WRITE_SINGLE_IO_REPLY` message types.

Note that this PR contributes only the data structures; there is no way to interact with the IO services through the `motoman_driver` nodes yet.

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
I know I'm not helping #92 by not immediately including tests, but those can/will be contributed later.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, thanks for this.  I'm going to get some critical PRs out of the way (hopefully today) and then do a release.  After the release, I'll come back around to this PR.

**Comment by gavanderhoorn** [comment]
Just a note: these classes have been extensively used the past couple of months. So although no real unit-tests are included in this PR, the large amount of runtime testing we've performed on them seems to indicate that they are alright.

Not to say that real unit-tests are sorely needed, but I already pointed that out in #92.

**Comment by gavanderhoorn** [comment]
Rebased to trigger Travis.

File: motoman_driver/include/motoman_driver/simple_message/messages/motoman_read_single_io_message.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  // Overrides - SimpleSerialize
+  bool load(industrial::byte_array::ByteArray *buffer);
+  bool unload(industrial::byte_array::ByteArray *buffer);
+
+  unsigned int byteLength()
+  {
+    return this->cmd_.byteLength();
+  }
+
+  motoman::simple_message::io_ctrl::ReadSingleIO cmd_;
```
Comment (mmoerdijk) [inline motoman_driver/include/motoman_driver/simple_message/messages/motoman_read_single_io_message.h:118]: Should this be private?

Comment (gavanderhoorn) [reply motoman_driver/include/motoman_driver/simple_message/messages/motoman_read_single_io_message.h:118]: Yes, that might make sense.

None of the other message classes have the wrapped data structure as a private member though, so if we want to maintain consistency we should probably leave it public.

I would welcome a PR that makes all wrapped structures private though.

**Comment by shaun-edwards** [comment]
+1...thanks for the contribution....merging.

**Pull request merged**